Demo/src/main/java/com/braintreepayments/demo/ShopperInsightsFragmentV2.kt

```diff
@@ -0,0 +1,298 @@
+                is PayPalPaymentAuthRequest.Failure -> {
+                    handleError(authRequest.error)
+                }
+
+                is PayPalPaymentAuthRequest.ReadyToLaunch -> {
+                    when (val paypalPendingRequest = paypalLauncher.launch(requireActivity(), authRequest)) {
+                        is PayPalPendingRequest.Started -> {
+                            paypalStartedPendingRequest = paypalPendingRequest
+                        }
+
+                        is PayPalPendingRequest.Failure -> {
+                            Toast.makeText(requireContext(), paypalPendingRequest.error.message, Toast.LENGTH_LONG)
+                                .show()
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun initShopperInsightsClient() {
+        fetchAuthorization { authResult ->
+            when (authResult) {
+                is BraintreeAuthorizationResult.Success -> {
+                    shopperInsightsClient = ShopperInsightsClientV2(requireContext(), authResult.authString)
+                    shopperInsightsClientSuccessfullyInstantiated = true
+                }
+                is BraintreeAuthorizationResult.Error -> {
+                    // Handle error, e.g., show error message
+                }
+            }
+        }
+    }
+
+    private fun handleCreateCustomerSession(emailText: String, countryCodeText: String, nationalNumberText: String) {
+        val customerSessionRequest = CustomerSessionRequest(
+            hashedEmail = emailText.sha256(),
+//            hashedPhoneNumber = nationalNumberText.sha256()
+        )
+        shopperInsightsClient.createCustomerSession(customerSessionRequest) { result ->
+            when (result) {
+                is CustomerSessionResult.Success -> {
+                    viewModel.sessionId.update { result.sessionId }
+                }
+                is CustomerSessionResult.Failure -> {
+                    // Handle failure, e.g., show error message
+                    val error = result.error
+                }
+            }
+        }
+    }
+
+    private fun handleUpdateCustomerSession(emailText: String, countryCodeText: String, nationalNumberText: String) {
+        val customerSessionRequest = CustomerSessionRequest(hashedEmail = emailText.sha256())
+        val requestId = "session-id-${UUID.randomUUID()}"
+        shopperInsightsClient.updateCustomerSession(customerSessionRequest, requestId) { result ->
+            when (result) {
+                is CustomerSessionResult.Success -> {
+                    viewModel.sessionId.update { result.sessionId }
+                }
+                is CustomerSessionResult.Failure -> {
+                    // Handle failure
+                }
+            }
+        }
+    }
+
+    private fun handleGetRecommendations(sessionId: String) {
+        val sessionId = "94f0b2db-5323-4d86-add3-paypal000000"
```

Comment:
Should we make this hardcoded sessionId the default value of the `sessionId` function param?